src/provider/provider_level_zero.c

```diff
@@ -350,12 +360,22 @@ umf_result_t umfLevelZeroMemoryProviderParamsSetResidentDevices(
         LOG_ERR("Level Zero memory provider params handle is NULL");
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
-
-    if (deviceCount && !hDevices) {
+    if (deviceCount > 0 && hDevices == NULL) {
         LOG_ERR("Resident devices array is NULL, but deviceCount is not zero");
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
+    for (uint32_t first_idx = 0; first_idx < deviceCount; first_idx++) {
+        for (uint32_t second_idx = 0; second_idx < first_idx; second_idx++) {
+            if (hDevices[first_idx] == hDevices[second_idx]) {
+                LOG_ERR("resident devices are not unique, idx:%u and "
+                        "idx:%u both point to device:%p",
```

Comment:
can you please add spaces before `%` (applies to all places writing `... :% ...` -> `... : % ...`)